src/loaders/genome_attributes/compute_genome_attribs.py

```diff
@@ -185,6 +185,41 @@ def _run_gtdb_tk_classify_wf(batch_file_path, work_dir, debug, genome_ids, progr
         f'{len(genome_ids)} genomes')
 
 
+def _map_tool_id_to_genome_id(tool_name, original_genome_id, genome_file_name):
+    """
+    Construct a dictionary that associates tool-generated genome IDs with the original genome IDs used to create
+    the _key field of a document.
+
+    The method is specific to the tool being used, as each tool has its own logic for generating unique genome IDs.
+    """
+
+    if tool_name == 'checkm2':
+        # CheckM2 uses the base name (without extension) of genome_file as the genome identifier
+        # We know for checkM2 extension 'genomic.fna.gz' is consistently used for genome files
+
+        base_name = genome_file_name.split('.fna.gz')[0]
+        return {base_name: original_genome_id}
+
+    elif tool_name == 'gtdb_tk':
+        # When creating the batch file for GTDB-TK, we use the genome ID as the identifier for each genome.
+        return {original_genome_id: original_genome_id}
+    else:
```

Comment:
Maybe not yet, but we might want to think about creating an interface that defines all the tool-specific methods like these and the run methods and implementing a class for each tool so that all the code for each tool is in one place and it's clear where code needs to be added to add a new tool

Reply:
👍 added todo item.

Reply:
:+1: